enumerables.rb

```diff
@@ -0,0 +1,126 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |x| yield self[x] }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |x| yield self[x], x }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    ret_arr = []
+    my_each do |x|
+      next unless yield(x)
+
+      ret_arr << x
+    end
+    ret_arr
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  def my_all?(pattern = nil)
+    return false if include?(nil) || include?(false)
+
+    my_each do |x|
+      if block_given?
+        return false unless yield x
+      elsif pattern.class == Regexp
+        return false unless pattern =~ x
+      elsif pattern.class == Class
+        return false unless x.class == pattern
+      else
+        return true
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    return true if include?(nil)
+    return true if include?(false)
+
+    my_each do |x|
+      if block_given?
+        return true if yield x
+      elsif pattern.class == Regexp
+        return true if pattern =~ x
+      elsif pattern.class == Class
+        return true if x.class == pattern
+      else
+        return false
+      end
+    end
+    false
+  end
```

Comment:
`my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, 2, 3].my_any? #should return true
[1, 2, nil].my_any? #should return true
[false, false, nil].my_any? #should return false
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
[1, 2, 2].my_any?(3)   #should return false
[true, false].my_any?(false)  #should return true
```

Reply:
Done! Thank you for letting me know. I added some extra conditions to make it return the correct values.